### ISSUE TITLE
Export CSV incorrect output

### ISSUE THREAD

**Issue by sgoodswen** [acadsamplesdb/acad_samples#46]
For Samples -> Export CSV, the output columns are different to the columns used to [Upload samples]. This means that if the CSV is imported into an Excel sheet  (with sheet name changed to sample), the [Upload samples] option generates an error

e.g. Cannot upload data: 'int' object has no attribute 'lower'

See attached file
[test_CSV.xlsx](https://github.com/acadsamplesdb/acad_samples/files/203946/test_CSV.xlsx)
 to recreate error.

The CSV header output is :

acad_num,other_num,group,category,extracted_by_date,acad_location,date_of_entry,abc_num,organism,common_name,genus,species,subspecies,notes,region,country,state,locality,lat,lon,datum,collection_date,collected_by,quality,museum,museum_num,sampled_by,sample_date,mismatch_reason,sample_repat

The [Upload samples] format is:

acad_num other_num  group   common_name genus   species subspecies  category    extracted_by_date   acad_location   date_of_entry   abc_num description region  country state   locallity   location    lat lon datum   collection_date collected_by    quality museum  museum_num  sampled_by  sample_date mismatch_reason sample_repat    import_permit_#

**Comment by Li-ReDBox** [comment]
I think these two are for different purposes. When uploading, probably just keep as less as possible number of columns is not a bad idea. But when exporting, you want more. 

@jsoubrier  could you please review the table below and let me know if they should be identical; if so, what is the set.

| Export | Upload |
| --- | --- |
| acad_num | acad_num |
| abc_num | abc_num |
| acad_location | acad_location |
| category | category |
| collected_by | collected_by |
| collection_date | collection_date |
| common_name | common_name |
| country | country |
| date_of_entry | date_of_entry |
| datum | datum |
|  | description |
| extracted_by_date | extracted_by_date |
| genus | genus |
| group | group |
|  | import_permit_# |
| lat | lat |
| locality | locallity |
|  | location |
| lon | lon |
| mismatch_reason | mismatch_reason |
| museum | museum |
| museum_num | museum_num |
| notes |  |
| organism |  |
| other_num | other_num |
| quality | quality |
| region | region |
| sample_date | sample_date |
| sample_repat | sample_repat |
| sampled_by | sampled_by |
| species | species |
| state | state |
| subspecies | subspecies |

**Comment by jsoubrier** [comment]
Thanks Li.
I think it makes sense if we make the Export set exactly matching the Upload one.
To me it looks like the Export scheme comes from an old set (outdated).
The only thing that I can see could be added to the Export set (at the end) is information corresponding to the color code in place. Has the sample been: 
Extracted?
Amplifed?
Enriched?
Sequenced?
Carbon dated?

If there is an easy way to lift over the info of the colored stars and symbols, we can make 5 more entries in the Export with either yes or no in them, otherwise, don't bother and just match the Upload set.

Cheers,
Ju

**Comment by Li-ReDBox** [comment]
I would think anything you see in the sample table should be able be extracted and exported. Do you have any example of having the extra information in a csv file? Five extra columns with yes no?

**Comment by jsoubrier** [comment]
Yes, I think 5 extra columns named "Extracted, Amplified, Enriched, Sequenced, C14", and YES/NO would be good.

**Comment by Li-ReDBox** [comment]
I will make export has all the columns in upload csv and five extra columns. I will talked to you if these extra columns deemed to hard. I hope not.

**Comment by Li-ReDBox** [comment]
@jsoubrier I have an old upload csv which looks having different columns listed here. Could you confirm if this is the complete and correct set?

acad_num other_num group common_name genus species subspecies category extracted_by_date acad_location date_of_entry abc_num description region country state locallity location lat lon datum collection_date collected_by quality museum museum_num sampled_by sample_date mismatch_reason sample_repat import_permit_#

**Comment by jsoubrier** [comment]
Yes, this is exactly the scheme with use for upload (except for the acad_num at the beginning, this column is removed for normal users as the DB is generating them - only admin users can force upload with acad_num which replaces the entries).

**Comment by Li-ReDBox** [comment]
I have not been involved in any design, so I have no knowledge of the functions and purposes of export or upload.

My observation is that the export function exports EXACTLY ALL the fields defined by the Sample model. By comparison, I suspect extra fields in the upload sheets are processed or ignored when uploading. I would say they are different functions not mirroring each other.

I have not investigated the re-uploading error because this issue asks to change export fields. When I gave the list of upload csv "fields" from above for exporting, as they are not in the model, error occurred. Particularly, `import_permit_#`, I have no idea what it is and where should it come from.

Please advise how to precede?

**Comment by jsoubrier** [comment]
Thanks for having a look at that Li,
I'll ask the group at lab meeting today if this function is critical for future work, and will let you know.
In the meantime, I would suggest leaving it as is for now.

**Comment by Li-ReDBox** [comment]
Thanks. will do that.
